integrations/huggingface-transformers/tutorials/question-answering/question_answering-cli.md

```diff
@@ -0,0 +1,372 @@
+
+The `question_answering` training script accepts JSON files in the form:
+
+```bash
+{
+  'data': [
+    {'question': 'What is my Name?', 'context': 'My name is Robert', "answers":{'answer_start':[11], 'text':['Robert']}},
+    {'question': 'What is my Name?', 'context': 'My name is Mark', "answers":{'answer_start':[11], 'text':['Mark']}},
+    {'question': 'What is my Name?', 'context': 'My name is Ben', "answers":{'answer_start':[11], 'text':['Ben']}},
+    ...
+  ]
+}
+```
+
+Run the following to convert the dataset to this format and dump to a json file.
+
+```python
+# load dataset
+from datasets import load_dataset
+squadshifts = load_dataset("squadshifts", "amazon")["test"].train_test_split(test_size=.2)
+
+# wrap dataset
+train_dict = {"data":[]}
+val_dict = {"data":[]}
+for row in squadshifts["train"]:
+  train_dict["data"].append(row)
+for row in squadshifts["test"]:
+  val_dict["data"].append(row)
+
+# dump to json files
+import json
+def dict_to_json_file(path, dictionary):
+  with open(path, 'w') as file:
+      json_string = json.dumps(dictionary, default=lambda o: o.__dict__, sort_keys=True, indent=2)
+      file.write(json_string)
+dict_to_json_file("squadshifts-train.json", train_dict)
+dict_to_json_file("squadshifts-val.json", val_dict)
+```
+
+#### **Kick off Training**
+
+To use the local files with the CLI, pass `--train_file squadshifts-train.json --validation_file squadshifts-val.json`:
+
+Run the following:
+```bash
+sparseml.transformers.train.question_answering \
+  --output_dir obert_base_pruned90_quant_squadshifts \
+  --recipe zoo:nlp/question_answering/obert-base/pytorch/huggingface/squad/pruned90_quant-none \
+  --recipe_args '{"num_epochs":8, "qat_start_epoch":4.0, "observer_epoch":7.0}' \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --distill_teacher dense_tracher \
+  --train_file squadshifts-train.json --validation_file squadshifts-val.json \
+  --do_train --do_eval --evaluation_strategy epoch --logging_strategy epoch --save_steps 1000 --preprocessing_num_workers 32 \
+  --per_device_train_batch_size 8 --per_device_eval_batch_size 32 --gradient_accumulation_steps 2 \
+  --max_seq_length 384 --doc_stride 128 \
+  --seed 42
+```
+
+Without doing any hyperparameter search, the script runs for 13 epochs and converges to ~70% F1 score.
+
+Note that in this case, we used the SQuAD transfer learning recipe (identified by 
+`zoo:nlp/question_answering/obert-base/pytorch/huggingface/squad/pruned90_quant-none`). Since the Squadshifts dataset is similiar to the SQuAD dataset, 
+we chose the same hyperparameters. To adjust the hyperparameters, either download the recipe and modify directly (passing
+a local file to the CLI) or pass an argument to `--recipe_args`.
+
+### Sparse Transfer Learning with a Custom Teacher
+
+To support the transfer learning process, we can apply model distillation, just like we did for the SQuAD case.
```

Comment:
We can apply model distillation to support the transfer learning process, just like we did for the SQuAD case.